sound/soc/sof/topology.c

```diff
@@ -3637,11 +3657,20 @@ static int sof_complete(struct snd_soc_component *scomp)
 				return ret;
 
 			swidget->complete = ret;
+
+			/* set pipe_widget for all widgets with same pipeline_id */
+			list_for_each_entry_reverse(comp_swidget, &sdev->widget_list, list)
+				if (comp_swidget->pipeline_id == swidget->pipeline_id) {
+					comp_swidget->pipe_widget = swidget;
```

Comment:
Do we need to set pipeline`s own pipe_widget to itself ?

Reply:
I also just wonder how widgets are stored in widget_list and list_for_each_entry_reverse will save time

Reply:
they are stored in the order they are parsed from topology but list_for_each_entry() and list_for_each_reverse() wont make a difference in tme here